rusk/src/bin/main.rs

```diff
@@ -79,66 +79,46 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         _ => unreachable!(),
     };
 
-    let _tempdir = match args.get_one::<PathBuf>("state_zip_file") {
+    let tempdir = match args.get_one::<PathBuf>("state_file") {
         Some(state_zip) => ephemeral::configure(state_zip)?,
         None => None,
     };
-
-    let router = {
-        let rusk = Rusk::new(rusk_profile::get_rusk_state_dir()?)?;
-
-        let kadcast = KadcastDispatcher::new(
-            config.kadcast.clone().into(),
-            config.kadcast_test,
-        )?;
-
-        let network =
-            NetworkServer::with_interceptor(kadcast, CompatibilityInterceptor);
-        let state =
-            StateServer::with_interceptor(rusk, CompatibilityInterceptor);
-        let prover = ProverServer::with_interceptor(
-            RuskProver::default(),
-            CompatibilityInterceptor,
+    let state_dir = rusk_profile::get_rusk_state_dir()?;
+    tracing::info!("Using state from {state_dir:?}");
+    let rusk = Rusk::new(state_dir)?;
+
+    // Set up a node where:
+    // transport layer is Kadcast with message ids from 0 to 255
+    // persistence layer is rocksdb
+    type Services = dyn LongLivedService<Kadcast<255>, rocksdb::Backend, Rusk>;
+
+    // Select list of services to enable
+    let service_list: Vec<Box<Services>> = vec![
+        Box::<MempoolSrv>::default(),
+        Box::new(ChainSrv::new(config.chain.consensus_keys_path())),
+        Box::new(DataBrokerSrv::new(config.databroker())),
+    ];
+
+    let db_path = tempdir
+        .as_ref()
+        .map_or_else(|| config.chain.db_path(), |t| t.path().to_path_buf());
+    let db = rocksdb::Backend::create_or_open(db_path);
+    let net = Kadcast::new(config.clone().kadcast.into());
+
+    let node = Node::new(net, db, rusk.clone());
+
+    let mut _ws_server = None;
+    if config.ws.listen {
+        _ws_server = Some(
+            WsServer::bind(rusk, node.clone(), config.ws.listen_addr()).await?,
         );
+    }
 
-        Server::builder()
-            .layer(RuskVersionLayer)
```

Comment:
By deprecating grpc service, do we intend to deprecate `dusk-blockchain` on merging this PR?

if not, how will dusk-blockchain execute VM calls from now on?

Reply:
Dusk-blockchain will target the previous rusk release v0.6.0 (with grpc)

Any core change (like contract, state-tree etc) will be applied to that longliving branch too until we fully deprecate dusk-blockchain

Reply:
By long-lived branch you mean the `node_refactor` branch will not be merged until we fully deprecate dusk-blockchain?

Reply:
No, a long living branch that will be created on top of 0.6.0 tag (that is the version of rusk exposing grpc)

Reply:
Out of curiosity, what were the concerns of having a `feature` in rusk binary to enable/disable grpc until we fully remove it? 

In the case of using a tag, is it possible to introduce a core change in one of the branches making dusk-blockchain and rusk incompatible?